lib/pbench/server/db/models/tracker.py

```diff
@@ -529,50 +620,51 @@ def __init__(self, **kwargs):
             key     Metadata key (from METADATA_KEYS)
             value   String value for key
         """
-        try:
-            key = kwargs.get("key")
-        except KeyError:
-            raise MetadataBadKey("Metadata key is required")
-        else:
-            if key not in Metadata.METADATA_KEYS:
-                raise MetadataBadKey(f"Metadata key {key} is not supported")
-        try:
-            value = kwargs.get("value")
-        except KeyError:
-            raise MetadataBadKey(f"Metadata key {key} value is required")
+        if "key" not in kwargs:
+            raise MetadataMissingParameter("key")
+        key = kwargs.get("key")
+        if key not in Metadata.METADATA_KEYS:
+            raise MetadataBadKey(key)
+        if "value" not in kwargs:
+            raise MetadataMissingKeyValue(key)
+        value = kwargs.get("value")
         super().__init__(**kwargs)
         self.key = key
         self.value = value
 
     @staticmethod
     def create(**kwargs):
+        if "dataset" not in kwargs:
+            raise MetadataMissingParameter("dataset")
+        dataset = kwargs.get("dataset")
+        if type(dataset) is not Dataset:
+            raise DatasetBadParameterType(dataset, Dataset)
+
         try:
             meta = Metadata(**kwargs)
-            meta.add(kwargs.get("dataset"))
-        except KeyError:
-            raise MetadataError("Metadata create missing argument 'dataset'")
+            meta.add(dataset)
         except Exception:
             return None
```

Comment:
Should this failure be logged here?  (Will the caller have enough context to log/diagnose the failure?)

Reply:
When I patched in logging through the `Database.Base` class, I did the `Dataset` operations to test it, but never got around to expanding that to `Metadata`. Working on that now... thanks.